tests/karma/unit/controllers/contacts.js

```diff
@@ -317,5 +331,89 @@ describe('Contacts controller', () => {
+        scrollLoaderCallback();
+        assert.equal(searchService.args[1][2].skip, 50);
+        assert.equal(searchService.args[1][2].limit, 50);
+      });
+    });
+
+    it('when paginating, does modify skip for non-admins #4085', () => {
+      const searchResult = { _id: 'search-result' };
+      searchResults = Array(50).fill(searchResult);
+
+      return createController().getSetupPromiseForTesting({ scrollLoaderStub }).then(() => {
+        const lhs = contactsLiveList.getList();
+        assert.equal(lhs.length, 51);
+        scrollLoaderCallback();
+        assert.equal(searchService.args[1][2].skip, 50);
+        assert.equal(searchService.args[1][2].limit, 50);
+      });
+    });
+
+    it('when refreshing list as admin, does not modify limit #4085', () => {
+      isAdmin = true;
+      userSettings = KarmaUtils.promiseService(null, { facility_id: undefined });
+      const searchResult = { _id: 'search-result' };
+      searchResults = Array(10).fill(searchResult);
+
+      return createController().getSetupPromiseForTesting({ scrollLoaderStub }).then(() => {
+        const lhs = contactsLiveList.getList();
+        changesCallback();
+        assert.equal(lhs.length, 10);
+        assert.equal(searchService.args[1][2].limit, 10);
+        assert.equal(searchService.args[1][2].skip, undefined);
+      });
+    });
+
+    it('when refreshing list as non-admin, does modify limit #4085', () => {
+      const searchResult = { _id: 'search-result' };
+      searchResults = Array(10).fill(searchResult);
+
+      return createController().getSetupPromiseForTesting({ scrollLoaderStub }).then(() => {
+        const lhs = contactsLiveList.getList();
+        assert.equal(lhs.length, 11);
+        changesCallback();
+        assert.equal(searchService.args[1][2].limit, 10);
+        assert.equal(searchService.args[1][2].skip, undefined);
+      });
+    });
+
+    it('resets limit/skip modifier when filtering #4085', (done) => {
+      let lhs;
+      const searchResult = { _id: 'search-result' };
+      searchResults = Array(10).fill(searchResult);
+
+      createController().getSetupPromiseForTesting({ scrollLoaderStub }).then(() => {
+        lhs = contactsLiveList.getList();
+        assert.equal(lhs.length, 11);
+        scope.filters = { search: true };
+        searchResults = Array(50).fill(searchResult);
+        searchService.returns(Promise.resolve(searchResults));
+        scope.search();
+        assert.equal(searchService.args[1][2].limit, 50);
+        assert.equal(searchService.args[1][2].skip, undefined);
+        setTimeout(() => {
+          lhs = contactSearchLiveList.getList();
+          assert.equal(lhs.length, 50);
+          //aand paginate the search results, also not skipping the extra place
+          scrollLoaderCallback();
+          assert.equal(searchService.args[2][2].skip, 50);
+          assert.equal(searchService.args[2][2].limit, 50);
+          done();
+        });
+      }).catch(e => {
```

Comment:
I don't think you need this `catch` if you return the promise that `createController()` returns. You also don't need the `done` parameter. Essentially if you return a promise from a test the running listens for `then` and passes the test and catches errors and fails the test.

Reply:
I have updated the tests so they use `Promise.resolve()` instead of `setTimeout` to simulate Search service responses.